pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/DefaultRebalancePreChecker.java

```diff
@@ -279,6 +299,48 @@ private RebalancePreCheckerResult checkDiskUtilization(Map<String, Map<String, S
         : RebalancePreCheckerResult.error(message.toString());
   }
 
+  private RebalancePreCheckerResult checkRebalanceConfig(RebalanceConfig rebalanceConfig, TableConfig tableConfig,
+      Map<String, Map<String, String>> currentAssignment, Map<String, Map<String, String>> targetAssignment) {
+    StringBuilder message = new StringBuilder();
+    boolean pass = true;
+    if (rebalanceConfig.isBestEfforts()) {
+      pass = false;
+      message.append("bestEfforts is enabled, only enable it if you know what you are doing\n");
+    }
+    List<String> segmentsToMove = SegmentAssignmentUtils.getSegmentsToMove(currentAssignment, targetAssignment);
+
+    int numReplicas = Integer.MAX_VALUE;
+    for (String segment : segmentsToMove) {
+      numReplicas = Math.min(targetAssignment.get(segment).size(), numReplicas);
+    }
+
+    if (rebalanceConfig.isDowntime() && !segmentsToMove.isEmpty() && numReplicas > 1) {
+      pass = false;
+      message.append("Number of replicas (")
+          .append(numReplicas)
+          .append(") is greater than 1, downtime is not recommended.\n");
+    }
+    if (rebalanceConfig.getMinAvailableReplicas() >= 0) {
+      // For non-negative value, use it as min available replicas
+      if (rebalanceConfig.getMinAvailableReplicas() >= numReplicas) {
+        pass = false;
+        message.append("The number of replicas (")
+            .append(numReplicas)
+            .append(") is less than minAvailableReplicas. The rebalance would fail.\n");
+      }
+    }
+    if (!rebalanceConfig.isIncludeConsuming() && tableConfig.getTableType() == TableType.REALTIME) {
+      pass = false;
+      message.append("includeConsuming is disabled for a realtime table.\n");
+    }
+    if (rebalanceConfig.isBootstrap()) {
+      pass = false;
+      message.append("bootstrap is enabled, only enable it if you know what you are doing\n");
```

Comment:
nit: let's reword this: "bootstrap is enabled which can cause a large amount of data movement, double check if this is intended"